pkg/resources/api_integration.go

```diff
@@ -0,0 +1,340 @@
+				return err
+			}
+		case "API_BLOCKED_PREFIXES":
+			if val := v.(string); val != "" {
+				if err = d.Set("api_blocked_prefixes", strings.Split(val, ",")); err != nil {
+					return err
+				}
+			}
+		case "API_AWS_IAM_USER_ARN":
+			if err = d.Set("api_aws_iam_user_arn", v.(string)); err != nil {
+				return err
+			}
+		case "API_AWS_ROLE_ARN":
+			if err = d.Set("api_aws_role_arn", v.(string)); err != nil {
+				return err
+			}
+		case "API_AWS_EXTERNAL_ID":
+			if err = d.Set("api_aws_external_id", v.(string)); err != nil {
+				return err
+			}
+		case "AZURE_CONSENT_URL":
+			if err = d.Set("azure_consent_url", v.(string)); err != nil {
+				return err
+			}
+		case "AZURE_MULTI_TENANT_APP_NAME":
+			if err = d.Set("azure_multi_tenant_app_name", v.(string)); err != nil {
+				return err
+			}
+		default:
+			log.Printf("[WARN] unexpected api integration property %v returned from Snowflake", k)
+		}
+	}
+
+	return err
+}
+
+// UpdateAPIIntegration implements schema.UpdateFunc
+func UpdateAPIIntegration(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	id := d.Id()
+
+	stmt := snowflake.ApiIntegration(id).Alter()
+
+	// This is required in case the only change is to UNSET API_ALLOWED_PREFIXES.
+	var runSetStatement bool
+
+	if d.HasChange("comment") {
+		runSetStatement = true
+		stmt.SetString("COMMENT", d.Get("comment").(string))
+	}
+
+	if d.HasChange("enabled") {
+		runSetStatement = true
+		stmt.SetBool(`ENABLED`, d.Get("enabled").(bool))
+	}
+
+	if d.HasChange("api_allowed_prefixes") {
+		runSetStatement = true
+		stmt.SetStringList("API_ALLOWED_PREFIXES", expandStringList(d.Get("api_allowed_prefixes").([]interface{})))
+	}
+
+	// We need to UNSET this if we remove all api blocked prefixes. I don't think
+	// @TODO move the SQL back to the snowflake package
+	if d.HasChange("api_blocked_prefixes") {
+		v := d.Get("api_blocked_prefixes").([]interface{})
+		if len(v) == 0 {
+			err := snowflake.Exec(db, fmt.Sprintf(`ALTER API INTEGRATION %v UNSET API_BLOCKED_PREFIXES`, d.Id()))
+			if err != nil {
+				return fmt.Errorf("error unsetting api_blocked_prefixes: %w", err)
+			}
+		} else {
+			runSetStatement = true
+			stmt.SetStringList("API_BLOCKED_PREFIXES", expandStringList(v))
+		}
+	}
+
+	if d.HasChange("api_provider") {
+		runSetStatement = true
+		err := setAPIProviderSettings(d, stmt)
+		if err != nil {
+			return err
+		}
+	} else {
+		if d.HasChange("api_aws_role_arn") {
+			runSetStatement = true
+			stmt.SetString("API_AWS_ROLE_ARN", d.Get("api_aws_role_arn").(string))
+		}
+		if d.HasChange("azure_tenant_id") {
+			runSetStatement = true
+			stmt.SetString("AZURE_TENANT_ID", d.Get("azure_tenant_id").(string))
+		}
+		if d.HasChange("azure_ad_application_id") {
+			runSetStatement = true
+			stmt.SetString("AZURE_AD_APPLICATION_ID", d.Get("azure_ad_application_id").(string))
+		}
+	}
+
+	if runSetStatement {
+		if err := snowflake.Exec(db, stmt.Statement()); err != nil {
```

Comment:
Can you log the statement being run? Potentially large.

Reply:
There is one already (the `snowflake.Exec()` method has `log.Print("[DEBUG] exec stmt ", query)`).